wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -316,6 +321,19 @@ deriveJSON Serokell.defaultOptions ''Address
 instance Arbitrary Address where
   arbitrary = Address . fromString <$> elements ["DEADBeef", "123456"]
 
+-- | Whether an address is valid or not.
+newtype AddressValidity = AddressValidity { getValid :: Bool }
+  deriving (Eq, Show, Generic)
+
+deriveJSON Serokell.defaultOptions ''AddressValidity
+
+instance Arbitrary AddressValidity where
+  arbitrary = AddressValidity <$> elements [True, False]
```

Comment:
We don't need to specify the full spectrum of possible values, cannot we simply write `arbitrary = AddressValidity <$> arbitrary`? This will simply piggyback on the `Arbitrary` instance for `Bool`, which is basically the same as writing this rolled-out-by-hand instance 😁